### PR TITLE
Add file association for .qgs/.qgz on Windows

### PR DESCRIPTION
**Pull request by nirvn**
Taken from https://cmake.org/cmake/help/book/mastering-cmake/chapter/Packaging%20With%20CPack.html?highlight=zip#setting-file-extension-associations-with-nsis 

Testing [a link](ms-windows-store://pdp/?productid=XP99H3BCX4BW7F).

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 06b543692735391cfcacc8063f251bd32ae7177a: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5621-e69e5c455516f40f4f7abc0d2fa4f5fc91f43669-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5621-e69e5c455516f40f4f7abc0d2fa4f5fc91f43669-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5621-e69e5c455516f40f4f7abc0d2fa4f5fc91f43669-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5621-e69e5c455516f40f4f7abc0d2fa4f5fc91f43669-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5621-e69e5c455516f40f4f7abc0d2fa4f5fc91f43669-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5621-e69e5c455516f40f4f7abc0d2fa4f5fc91f43669-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5621-e69e5c455516f40f4f7abc0d2fa4f5fc91f43669-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5621-e69e5c455516f40f4f7abc0d2fa4f5fc91f43669-android-x86.apk)
</details>

File: cmake/Package.cmake [hunk 1]
```diff
@@ -35,6 +35,30 @@ if(WIN32)
...
+                         '' 'Edit Project'
+        WriteRegStr HKCR 'QField\\shell\\edit\\command' \\
+                         '' '$INSTDIR\\usr\\bin\\qfield.exe \"%1\"'
+        System::Call \\
+          'Shell32::SHChangeNotify(i 0x8000000, i 0, i 0, i 0)'
+        ")
+
+    set(CPACK_NSIS_EXTRA_UNINSTALL_COMMANDS "
+        DeleteRegKey HKCR '.qgs'
+        DeleteRegKey HKCR '.qgz'
```
Comment (nirvn) [inline cmake/Package.cmake:58]:
```suggestion
```

**Comment by nirvn** [comment]
We shouldn't delete .qgs and .qgz extension key, this could have been changed by e.g. installing QGIS and we don't want to mess with that.

This also follows Microsoft's own recommendation not to delete that key (https://learn.microsoft.com/en-us/windows/win32/shell/fa-file-types).

**Comment by m-kuhn** [comment]
We already have a copy of the NSIS template in place, it's probably easier to just edit that in place https://github.com/opengisch/QField/blob/c87bcf667851fce93357e804e2b440f0fc70f125/cmake/NSIS.template.in#L752

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**